src/text_manipulation.py

```diff
@@ -41,43 +41,45 @@ def print_list(self, input_list):
      #return sentences_list
 
     def reorder_sentences(self, sentences_list):
+        # the below function takes a random sentence in a paragraph and replaces it a random different point
 
-        import random
+        # the line below finds the length of the paragraph
+        length_of_paragraph = len(sentences_list)
 
-        paragraph_list = list(sentences_list)
-        print(paragraph_list)
+        # the line below creates a copy of the original list of text
+        copy_sentences_list = sentences_list.copy()
 
-        a = len(paragraph_list)
-        print(a)
+        # the below takes a sentence from the length of the paragraph list
+        rand_idx = random.randrange(length_of_paragraph)
+        random_sentence = sentences_list[rand_idx]
 
-        copy_paragraph_list = paragraph_list.copy()
+        # the line below takes random sentence and indexes it to show the position of the random sentence was within the paragraph, it calls this variable 'index_random_sentence'
+        index_random_sentence = sentences_list.index(random_sentence)
```

Comment:
I thought i needed a copy of it because i need an unchanged version? You will have to talk me through this one.

Reply:
You already know the index of the list because you wrote the code that defined it yourself on line 53, because you used it on line 54 to remove the reandom_sentence. So line 57 is completely redundant. I think I might have gone a but too hardcore about mutibility, it's not really that big a thing in coding, just something you need to be aware of to avoid silly little (but very easy to make!) mistakes